### PR TITLE
Added custom Directory Uploads

### PR DESCRIPTION
**Pull request by jnthn-b**
In this Change the file loads the directory value from the config file and generates an string ($target_dir) out of that. 
This String is now the replacement for the hard coded value '/u/'.

### CODE REVIEW

**Comment by JoeGandy** [comment]
I think its worth combining these PRs as you have two separate ones to implement one piece of logic

**Comment by jnthn-b** [comment]
Can u merge the Pull Request?

**Comment by JoeGandy** [comment]
I haven't got time at the moment to review the change and make sure its okay, it might need some work.

In the mean time updating the documentation on the README is worth doing as part of the PR too

**Comment by jnthn-b** [comment]
Okay, will do that soon!

**Comment by jnthn-b** [comment]
Done!

**Comment by untuned** [comment]
bumping, no reply in 22 days

**Comment by JoeGandy** [comment]
Apologies, dropped the ball on this one.

It looks good! I have two concerns that are worth discussing:

- When the user changes the directory,i do we need to think about migrating existing content?
- When you change the directory, it kind of assumes the folder exists will this cause some errors on trying to upload a file to a folder that doesn't exist?

Again, sorry for the delay!

File: src/upload.php [hunk 1]
```diff
@@ -17,23 +17,29 @@
 
         $first_run = true;
         $files_exist_counter = 0;
-
+        
+        if ($config['upload_to_root']) {
+            $target_dir = '/';
+        }else {
+            $target_dir = '/'. $config['directory_for_uploads'] . '/';
```
Comment (JoeGandy) [inline src/upload.php]: Perhaps here we have a check to see if that directory exists, if it doesn't spit out an error?

**Comment by untuned** [comment]
Bumping and mentioning @jnthn-b.

**Comment by jnthn-b** [comment]
Might take a look at this soon, had to do a lot of things recently.

**Comment by jnthn-b** [comment]
okay, I just updated this. That *should* work.

**Comment by jnthn-b** [comment]
okay there seems to be a conflict with another pull request since i didnt updated this very regulary. Im going to take a look at that too.

**Comment by jnthn-b** [comment]
Im thinking of closing this PR due to the changes of the 'u' folder which is a big part of this at this point. So if this would go to continue the user would have to rename this folder to whatever he wants. I think an option in the Webinterface would be the way to go with this.

**Comment by JoeGandy** [comment]
Grew beyond initial issue, I think a fresh look at this would do wonders, so going to close this pr

**Comment by jnthn-b** [comment]
Yeah totally agree on that!

**Pull request closed without merging**